src/librustc_trans/trans/datum.rs

```diff
@@ -496,8 +509,10 @@ impl<'tcx> Datum<'tcx, Rvalue> {
 
             ByValue => {
                 lvalue_scratch_datum(
-                    bcx, self.ty, name, scope, self,
+                    bcx, self.ty, name, InitAlloca::Dropped, scope, self,
                     |this, bcx, llval| {
+                        debug!("populate call for Datum::to_lvalue_datum_in_scope \
+                                self.ty={:?}", this.ty);
                         call_lifetime_start(bcx, llval);
```

Comment:
From a quick glance a the dataflow calculation in the stack coloring pass, it looks like that might cause it to think that the alloca is dead when entering the basic block that has the second lifetime start. I didn't verify that though, because even if that does not happen, the lifetime will be extended to the entry block, so we could as well remove the second lifetime start call. That will reduce the potential for stack slot sharing, but I can't even guess how much of an impact that would have.